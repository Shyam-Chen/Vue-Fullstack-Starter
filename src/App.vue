<template>
  <v-app>
    <template v-if="!$route.meta.standalone">
      <v-navigation-drawer v-if="!$route.meta.home || isMobile" v-model="app$.drawer" :clipped="$vuetify.breakpoint.mdAndUp" app>
        <v-toolbar class="hidden-md-and-up" flat>
          <v-toolbar-title class="o-toolbar-title">
            <div class="o-router-link" @click="backToHome">
              <img src="/assets/images/logo.svg" alt="Logo" width="40" height="40" class="o-toolbar-image">
              <span class="primary--text">Oh My Vue</span>
            </div>
          </v-toolbar-title>
        </v-toolbar>

        <v-text-field v-if="!$route.meta.home || isMobile" solo append-icon="search" label="Search" class="hidden-md-and-up ma-3 o-text-field"></v-text-field>

        <v-divider></v-divider>

        <v-list nav dense>
          <template v-for="item in app$.navigation">
            <v-divider v-if="item.divider" :key="item.subheader + 'Divider'"></v-divider>
            <v-subheader v-if="item.subheader" :key="item.subheader">
              {{ $t(item.subheader) }}
            </v-subheader>

            <!-- if children -->
            <v-list-group v-if="item.children" :key="item.text" :prepend-icon="item.icon" :append-icon="icons.mdiChevronDown" :disabled="item.disabled">
              <template v-slot:activator>
                <v-list-item-content>
                  <v-list-item-title>{{ $t(item.text) }}</v-list-item-title>
                </v-list-item-content>
              </template>

              <template v-for="child in item.children">
                <!-- if sub-children -->
                <v-list-group v-if="child.children" :key="child.text" :prepend-icon="icons.mdiMenuDown" :disabled="child.disabled" sub-group>
                  <template v-slot:activator>
                    <v-list-item-content>
                      <v-list-item-title>{{ $t(child.text) }}</v-list-item-title>
                    </v-list-item-content>
                  </template>

                  <v-list-item v-for="subchild in child.children" :key="subchild.text" :to="`/${$route.params.lang}` + subchild.route" :disabled="subchild.disabled" ripple>
                    <v-list-item-action>
                      <v-icon>{{ subchild.icon }}</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                      <v-list-item-title>{{ $t(subchild.text) }}</v-list-item-title>
                    </v-list-item-content>
                  </v-list-item>
                </v-list-group>

                <!-- else not sub-children -->
                <v-list-item v-else :key="child.text" :to="`/${$route.params.lang}` + child.route" :disabled="child.disabled" ripple>
                  <v-list-item-action>
                    <v-icon>{{ child.icon }}</v-icon>
                  </v-list-item-action>
                  <v-list-item-content>
                    <v-list-item-title>{{ $t(child.text) }}</v-list-item-title>
                  </v-list-item-content>
                </v-list-item>
              </template>
            </v-list-group>

            <!-- else not children -->
            <v-list-item v-else :key="item.text" :to="`/${$route.params.lang}` + item.route" :href="item.href" :disabled="item.disabled" ripple>
              <v-list-item-action>
                <v-icon>{{ item.icon }}</v-icon>
              </v-list-item-action>
              <v-list-item-content>
                <v-list-item-title>{{ $t(item.text) }}</v-list-item-title>
              </v-list-item-content>
            </v-list-item>
          </template>
        </v-list>
      </v-navigation-drawer>

      <v-app-bar :clipped-left="$vuetify.breakpoint.mdAndUp" :flat="$route.meta.home" class="primary darken-1" dark app>
        <v-toolbar-title class="o-toolbar-title">
          <v-app-bar-nav-icon v-if="!$route.meta.home || isMobile" @click.stop="app$.drawer = !app$.drawer">
            <v-icon>{{ icons.mdiMenu }}</v-icon>
          </v-app-bar-nav-icon>
          <router-link class="white--text o-router-link" :to="`/${$route.params.lang}`">
            <img src="/assets/images/logo.svg" alt="Logo" width="40" height="40" class="hidden-sm-and-down o-toolbar-image">
            <span>Oh My Vue</span>
          </router-link>
        </v-toolbar-title>

        <v-text-field v-if="!$route.meta.home" flat solo-inverted hide-details prepend-inner-icon="search" label="Search" class="hidden-sm-and-down"></v-text-field>

        <v-spacer></v-spacer>

        <v-toolbar-items v-if="$route.meta.home && !isMobile">
          <v-btn :to="`/${$route.params.lang}/overview`" text>
            {{ $t('overview') }}
          </v-btn>

          <v-menu open-on-hover offset-y>
            <template v-slot:activator="{ on }">
              <v-btn text v-on="on">
                {{ $t('examples') }}<v-icon dark>
                  arrow_drop_down
                </v-icon>
              </v-btn>
            </template>

            <v-list>
              <template v-for="(item, index) in app$.navigation">
                <v-list-item v-if="index > 0 && index < 12" :key="index" :to="`/${$route.params.lang}` + item.route">
                  <v-list-item-action>
                    <v-icon>{{ item.icon }}</v-icon>
                  </v-list-item-action>
                  <v-list-item-content>
                    <v-list-item-title>{{ $t(item.text) }}</v-list-item-title>
                  </v-list-item-content>
                </v-list-item>
              </template>
            </v-list>
          </v-menu>
        </v-toolbar-items>

        <v-btn aria-label="Theme" icon @click="setTheme(!$vuetify.theme.dark ? 'dark' : 'light')">
          <v-icon>{{ $vuetify.theme.dark ? icons.mdiLightbulb : icons.mdiLightbulbOutline }}</v-icon>
        </v-btn>

        <v-menu bottom left>
          <template v-slot:activator="{ on }">
            <v-btn aria-label="Language" icon v-on="on">
              <v-icon>{{ icons.mdiTranslate }}</v-icon>
            </v-btn>
          </template>

          <v-list>
            <v-list-item v-for="lang in app$.languages" :key="lang.key" @click="setLanguage(lang.key)">
              <v-list-item-title>{{ lang.label }}</v-list-item-title>
            </v-list-item>
          </v-list>
        </v-menu>

        <v-btn icon to="/authorization/login">
          <v-icon>{{ icons.mdiAccount }}</v-icon>
        </v-btn>
      </v-app-bar>

      <v-content>
        <v-fade-transition mode="out-in">
          <router-view></router-view>
        </v-fade-transition>

        <v-footer height="auto" class="pa-3 primary darken-1 white--text">
          <img src="/assets/images/logo.svg" alt="Logo" width="50" height="50">

          <v-layout :column="isMobile" fill-height justify-space-between align-center class="ml-3 mr-3">
            <div class="subheading">
              <a href="https://vuejs.org/" class="subheading pa-2 white--text o-router-link">Learn Vue</a>
            </div>
            <div class="caption">
              Current Version: 2.6.10
            </div>
            <div class="caption text-xs-center">
              Powered by Oh My Vue &copy;{{ new Date().getFullYear() }}. <br>
              Code licensed under an MIT-style License. <br>
              Documentation licensed under CC BY 4.0.
            </div>
          </v-layout>
        </v-footer>
      </v-content>
    </template>

    <template v-else>
      <v-fade-transition mode="out-in">
        <router-view></router-view>
      </v-fade-transition>
    </template>
  </v-app>
</template>

<script>
// @flow

import { mapActions } from 'vuex';
import {
  mdiAccount,
  mdiChevronDown,
  mdiLightbulb,
  mdiLightbulbOutline,
  mdiMenu,
  mdiMenuDown,
  mdiTranslate,
} from '@mdi/js';

import layout from '~/shared/composes/layout';

import { IApp } from './types';
import actions from './actions';

export default {
  mixins: [layout],
  data() {
    return {
      icons: {
        mdiAccount,
        mdiChevronDown,
        mdiLightbulb,
        mdiLightbulbOutline,
        mdiMenu,
        mdiMenuDown,
        mdiTranslate,
      },
    };
  },
  computed: {
    app$(): IApp {
      return this.$store.state;
    },
  },
  created() {
    this.initialLanguage();
  },
  methods: {
    ...mapActions(Object.keys(actions)),
  },
};
</script>

<style scoped>
.o-toolbar-title {
  width: 20rem;
}

.o-router-link {
  text-decoration: none;
  cursor: pointer;
}

.o-toolbar-image {
  vertical-align: middle;
}

.o-text-field {
  height: 3.5rem;
}

.o-list-item-avatar {
  font-size: 2rem;
}
</style>
