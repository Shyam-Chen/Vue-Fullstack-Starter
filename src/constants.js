// @flow

import { IApp } from './types';

export const INITIAL: IApp = {
  navigation: [
    /** @name practical-examples */
    { icon: 'pageview', text: 'overview', route: '/overview', subheader: 'practicalExamples' },
    { icon: 'face', text: 'helloWorld', route: '/hello-world' },
    {
      icon: 'loyalty',
      text: 'guide',
      route: '/guide',
      children: [
        { text: 'templateRendering', route: '/guide/template-rendering' },
        {
          text: 'components',
          children: [
            { text: 'props', route: '/guide/components/props' },
            { text: 'slots', route: '/guide/components/slots' },
            { text: 'refs', route: '/guide/components/refs' },
            { text: 'customEvents', route: '/guide/components/custom-events' },
            { text: 'extends', route: '/guide/components/extends', disabled: true },
            { text: 'dynamic', route: '/guide/components/dynamic' },
            { text: 'recursion', route: '/guide/components/recursion' },
          ],
        },
        { text: 'lifecycleHooks', route: '/guide/lifecycle-hooks' },
        { text: 'transitions', route: '/guide/transitions' },
        { text: 'mixins', route: '/guide/mixins' },
        { text: 'directives', route: '/guide/directives' },
        { text: 'dependencyInjection', route: '/guide/dependency-injection' },
        { text: 'filters', route: '/guide/filters' },
        {
          text: 'routing',
          children: [
            { text: 'dynamicMatching', route: '/guide/routing/dynamic-matching' },
            { text: 'nestedRoutes', route: '/guide/routing/nested-routes' },
            { text: 'guards', route: '/guide/routing/guards' },
          ],
        },
        { text: 'stateManagement', route: '/guide/state-management' },
      ],
    },
    {
      icon: 'business',
      text: 'crudOperations',
      route: '/crud-operations',
      children: [
        { text: 'basic', route: '/crud-operations/basic' },
        { text: 'rest', route: '/crud-operations/rest' },
        { text: 'graphql', route: '/crud-operations/graphql' },
      ],
    },
    {
      icon: 'contacts',
      text: 'controls',
      route: '/controls',
      children: [
        { text: 'textFields', route: '/controls/text-fields' },
        { text: 'selects', route: '/controls/selects' },
        { text: 'selectionControls', route: '/controls/selection-controls' },
        { text: 'pickers', route: '/controls/pickers' },
        { text: 'autocomplete', route: '/controls/autocomplete' },
        { text: 'stepper', route: '/controls/stepper' },
        { text: 'tabs', route: '/controls/tabs', disabled: true },
        { text: 'treeView', route: '/controls/tree-view' },
        { text: 'calendar', route: '/controls/calendar' },
        { text: 'markdownEditor', route: '/controls/markdown-editor' },
        { text: 'richTextEditor', route: '/controls/rich-text-editor', disabled: true },
      ],
    },
    {
      icon: 'fas fa-table',
      text: 'dataTable',
      route: '/data-table',
      children: [
        { text: 'displayRows', route: '/data-table/display-rows' },
        { text: 'inlineEdit', route: '/data-table/inline-edit', disabled: true },
        { text: 'paginator', route: '/data-table/paginator' },
        { text: 'iterator', route: '/data-table/iterator' },
        { text: 'exportReport', route: '/data-table/export-report', disabled: true },
      ],
    },
    { icon: 'language', text: 'internationalization', route: '/internationalization' },
    { icon: 'fas fa-chart-bar', text: 'dataVisualization', route: '/data-visualization' },
    { icon: 'verified_user', text: 'authorization', route: '/authorization' },
    { icon: 'far fa-credit-card', text: 'payment', route: '/payment' },
    {
      icon: 'message',
      text: 'messages',
      route: '/messages',
      children: [
        { text: 'chatRoom', route: '/messages/chat-room' },
        { text: 'email', route: '/messages/email', disabled: true },
        { text: 'cloudMessaging', route: '/messages/cloud-messaging', disabled: true },
      ],
    },
    { icon: 'fas fa-cogs', text: 'machineLearning', route: '/machine-learning' },
  ],
  drawer: false || null,
  languages: [
    { key: 'en', label: 'English' },
    { key: 'zh-CN', label: '中文 (简体)' },
    { key: 'zh-TW', label: '中文 (正體)' },
    { key: 'ja', label: '日本語' },
    { key: 'ko', label: '한국어' },
  ],
};
