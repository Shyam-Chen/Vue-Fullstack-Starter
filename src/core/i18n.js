import { createStore } from 'vuex';
import mi from 'message-interpolation';

import enUS from '~/locales/en-US';

export const langList = {
  'en-US': 'American English',
  'ja-JP': '日本語',
  'zh-TW': '正體中文',
};

export const getUserLang = () => {
  const langListKeys = Object.keys(langList);

  if (langListKeys.includes(navigator.language)) {
    return navigator.language;
  }

  const liteLangListKeys = langListKeys.map(lang => lang.split(/[-]/)[0]);
  const ISO_639_1 = navigator.language.split(/[-]/)[0];

  if (liteLangListKeys.includes(ISO_639_1)) {
    for (let i = 0; i < langListKeys.length; i += 1) {
      const lang = langListKeys[i];

      if (lang.startsWith(ISO_639_1)) {
        return lang;
      }
    }
  }

  return 'en-US';
};

export const injectLanguage = (state, { lang, locale }) => {
  state.lang = lang;
  state.locale = locale;
};

export const i18n = createStore({
  state: {
    lang: 'en-US',
    locale: enUS,
  },
  actions: {
    async initialLanguage({ commit }, mod) {
      const lang = localStorage.getItem('lang') || getUserLang();
      document.documentElement.lang = lang;
      // TODO: const response = await import(`~/locales/${lang}.js`);
      const response = await import(`../locales/${lang}.js`);
      commit('injectLanguage', { lang, locale: response.default });

      if (mod) {
        const res = await import(`../modules/${mod}/locales/${lang}.js`);
        commit(`${mod}/injectLanguage`, { lang, locale: response.default });
      }
    },
    async setLanguage({ commit }, lang) {
      document.documentElement.lang = lang;
      localStorage.setItem('lang', lang);
      const response = await import(`../lang/${lang}.js`);
      commit('injectLanguage', { lang, locale: response.default });

      if (router.currentRoute.name !== 'singleSignOn') {
        const mod = router.currentRoute.name.split('/')[0];
        const res = await import(`../modules/${mod}/lang/${lang}.js`);
        commit(`${mod}/injectLanguage`, { lang, locale: response.default });
      }
    },
  },
  mutations: {
    injectLanguage,
  },
});

export const plugin = {
  install(app, options) {
    app.config.globalProperties.$f = mi;
  },
};
