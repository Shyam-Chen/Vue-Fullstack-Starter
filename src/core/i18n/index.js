import Vue from 'vue';
import I18n from 'vue-i18n';

import en from './en';

Vue.use(I18n);

export const userLang = () => {
  if (['zh-CN', 'zh-SG'].includes(navigator.language)) return 'zh-Hans';
  if (['zh-TW', 'zh-HK'].includes(navigator.language)) return 'zh-Hant';
  if (['pt-BR'].includes(navigator.language)) return 'pt-BR';
  return navigator.language.split(/[-]/)[0];
};

const i18n = new I18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages: { en },
});

export default i18n;
