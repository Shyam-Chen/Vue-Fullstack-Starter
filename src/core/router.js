import Vue from 'vue';
import Router from 'vue-router';
import RouterPrefetch from 'vue-router-prefetch';
import Meta from 'vue-meta';
import Analytics from 'vue-analytics';

import Home from '~/home/Home';
import NotFound from '~/not-found/NotFound';

import guide from '~/shell/guide/routes';

Vue.use(Router);
Vue.use(RouterPrefetch);
Vue.use(Meta);

const router = new Router({
  mode: 'history',
  base: process.env.APP_BASE,
  routes: [
    { path: '/', component: Home, meta: { home: true } },

    // { path: '/dashboard', component: () => import('~/dashboard/Dashboard') },

    /** @name overview */
    { path: '/overview', component: () => import('~/shell/overview/Overview') },

    /** @name hello-world */
    { path: '/hello-world', component: () => import('~/shell/hello-world/HelloWorld') },

    /** @name guide */
    ...guide,

    /** @name crud-operations */
    { path: '/crud-operations', redirect: '/crud-operations/basic' },
    { path: '/crud-operations/basic', component: () => import('~/shell/crud-operations/basic/Basic') },
    { path: '/crud-operations/rest', component: () => import('~/shell/crud-operations/rest/REST') },
    { path: '/crud-operations/graphql', component: () => import('~/shell/crud-operations/graphql/GraphQL') },

    /** @name controls */
    { path: '/controls', redirect: '/controls/text-fields' },
    { path: '/controls/text-fields', component: () => import('~/shell/controls/text-fields/TextFields') },
    { path: '/controls/selects', component: () => import('~/shell/controls/selects/Selects') },
    { path: '/controls/selection-controls', component: () => import('~/shell/controls/selection-controls/SelectionControls') },
    { path: '/controls/pickers', component: () => import('~/shell/controls/pickers/Pickers') },
    { path: '/controls/autocomplete', component: () => import('~/shell/controls/autocomplete/Autocomplete') },
    { path: '/controls/stepper', component: () => import('~/shell/controls/stepper/Stepper') },
    { path: '/controls/tree-view', component: () => import('~/shell/controls/tree-view/TreeView') },
    { path: '/controls/calendar', component: () => import('~/shell/controls/calendar/Calendar') },
    { path: '/controls/markdown-editor', component: () => import('~/shell/controls/markdown-editor/MarkdownEditor') },

    /** @name data-table */
    { path: '/data-table', redirect: '/data-table/display-rows' },
    { path: '/data-table/display-rows', component: () => import('~/shell/data-table/display-rows/DisplayRows') },
    { path: '/data-table/paginator', component: () => import('~/shell/data-table/paginator/Paginator') },
    { path: '/data-table/iterator', component: () => import('~/shell/data-table/iterator/Iterator') },

    /** @name internationalization */
    { path: '/internationalization', component: () => import('~/shell/internationalization/Internationalization') },

    /** @name data-visualization */
    { path: '/data-visualization', component: () => import('~/shell/data-visualization/DataVisualization') },

    /** @name authorization */
    { path: '/authorization', component: () => import('~/shell/authorization/Authorization') },
    { path: '/authorization/login', component: () => import('~/shell/authorization/Login'), meta: { standalone: true } },
    // { path: '/authorization/register', component: () => import('~/shell/authorization/Register'), meta: { standalone: true } },

    /** @name payment */
    { path: '/payment', component: () => import('~/shell/payment/Payment') },

    /** @name messages */
    { path: '/messages', redirect: '/messages/chat-room' },
    { path: '/messages/chat-room', component: () => import('~/shell/messages/chat-room/ChatRoom') },

    /** @name machine-learning */
    { path: '/machine-learning', component: () => import('~/shell/machine-learning/MachineLearning') },

    { path: '*', component: NotFound, meta: { standalone: true } },
  ],
  async scrollBehavior(to, from, savedPosition) {
    if (document.readyState !== 'complete') {
      await new Promise((resolve) => {
        const callback = () => {
          window.requestAnimationFrame(resolve);
          window.removeEventListener('load', callback);
        };

        window.addEventListener('load', callback);
      });
    }

    if (to.hash) return { selector: to.hash };
    if (savedPosition) return savedPosition;

    return new Promise((resolve) => {
      setTimeout(() => resolve({ y: 0 }), 200);
    });
  },
});

router.beforeResolve((to, from, next) => {
  if (to.name) {
    // progress.start();
  }

  next();
});

router.afterEach(() => {
  // progress.done();
});

Vue.use(Analytics, {
  id: process.env.GOOGLE_ANALYTICS || 'UA-XXXXXXXX-X',
  router,
  autoTracking: {
    page: process.env.NODE_ENV === 'production',
  },
});

export default router;
