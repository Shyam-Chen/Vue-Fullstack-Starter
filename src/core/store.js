import { reactive, provide, inject } from 'vue';

const state = {
  foo: 'foo',
  bar: 'bar',
};

export const stateSymbol = Symbol('state');
export const createState = () => reactive(state);
export const useState = () => inject(stateSymbol);
export const provideState = () => provide(stateSymbol, createState());
