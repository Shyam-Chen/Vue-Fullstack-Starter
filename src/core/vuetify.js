import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import colors from 'vuetify/lib/util/colors';
import webfontloader from 'webfontloader';
import LRU from 'lru-cache';

Vue.use(Vuetify);

webfontloader.load({
  google: {
    families: [
      'Roboto:100,300,400,500,700,900',
      // 'Roboto+Mono:500',
      // 'Material+Icons',
    ],
  },
  // custom: {
  //   families: [
  //     'Material Design Icons', // @mdi/js
  //     'Font Awesome 5',
  //   ],
  //   urls: [
  //     'https://cdn.jsdelivr.net/npm/@mdi/font@4.x/css/materialdesignicons.min.css',
  //     'https://use.fontawesome.com/releases/v5.0.8/css/all.css',
  //   ],
  // },
});

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    dark: localStorage.getItem('theme') === 'dark' || false,
    themes: {
      light: {
        primary: colors.blue.darken1,
        secondary: colors.blue.lighten1,
        accent: colors.pink,
      },
      dark: {
        primary: colors.blue.darken1,
        secondary: colors.blue.lighten1,
        accent: colors.pink,
      },
    },
    options: {
      minifyTheme(css) {
        return css.replace(/[\r\n|\r|\n]/g, '');
      },
      themeCache: new LRU({
        max: 10,
        maxAge: 60 * 60 * 1000,
      }),
    },
  },
});
