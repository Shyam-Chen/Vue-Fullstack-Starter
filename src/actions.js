// @flow

import { ActionContext } from 'vuex';

import router from '~/core/router';
import vuetify from '~/core/vuetify';
import i18n from '~/core/i18n';

import { IApp } from './types';
import { INITIAL } from './constants';

export default {
  setTheme({ state }: ActionContext<IApp>, val: string): void {
    state.theme = val;
    localStorage.setItem('theme', val);
    vuetify.framework.theme.dark = val === 'dark';
  },
  setLanguage(context: ActionContext<IApp>, val: string): void {
    import(`~/core/i18n/${val}`).then(
      (data): void => {
        i18n.setLocaleMessage(val, data.default);
        i18n.locale = val;
        document.documentElement.lang = val;
        sessionStorage.setItem('lang', val);
        router.push(`/${val}`);
      },
    );
  },
  initialLanguage({ dispatch }: ActionContext<IApp>): void {
    INITIAL.languages.forEach(
      ({ key }): void => {
        if (key === 'en') return;

        const lang = sessionStorage.getItem('lang');
        if (lang) dispatch('setLanguage', lang);

        if (navigator.language.includes(key)) {
          dispatch('setLanguage', key);
        }
      },
    );
  },
  backToHome({ state }: ActionContext<IApp>): void {
    router.push('/');
    state.drawer = false;
  },
};
