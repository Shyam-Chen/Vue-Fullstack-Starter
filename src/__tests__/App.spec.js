import { createLocalVue, shallowMount, RouterLinkStub } from '@vue/test-utils';
import Router from 'vue-router';
import Vuex from 'vuex';
import I18n from 'vue-i18n';
import mockdate from 'mockdate';

import store from '~/core/store';
import router from '~/core/router';
import i18n from '~/core/i18n';
import vuetify from '~/core/vuetify';

import Home from '~/home/Home';
import HelloWorld from '~/shell/hello-world/HelloWorld';
import StateManagement from '~/shell/guide/state-management/StateManagement';
import NotFound from '~/not-found/NotFound';

import App from '../App';
import { INITIAL as state } from '../constants';
import actions from '../actions';
import getters from '../getters';

const localVue = createLocalVue();

localVue.use(Router);
localVue.use(Vuex);
localVue.use(I18n);

jest.mock('~/shell/overview/Overview', () => ({
  name: 'Overview',
  render: h => h('div'),
}));
jest.mock('~/shell/hello-world/HelloWorld', () => ({
  name: 'HelloWorld',
  render: h => h('div'),
}));

jest.mock('~/shell/guide/state-management/StateManagement', () => ({
  name: 'StateManagement',
  render: h => h('div'),
}));

describe('App', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(App, {
      localVue,
      router,
      store,
      i18n,
      vuetify,
      stubs: {
        RouterLink: RouterLinkStub,
      },
      mocks: {
        Date: mockdate.set('2020-01-10'),
      },
    });
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();

    router.push('/guide/state-management');
    expect(wrapper.find(StateManagement).exists());
  });

  it('', () => {
    router.push('/hello-world');
    expect(wrapper.find(HelloWorld).exists());
  });
});
