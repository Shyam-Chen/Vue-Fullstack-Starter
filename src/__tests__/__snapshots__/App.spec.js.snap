// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render an initial component 1`] = `
<v-app-stub id="app">
  <!---->
  <v-app-bar-stub dark="true" tag="header" tile="true" extensionheight="48" flat="true" src="" value="true" app="true" clippedleft="true" class="primary darken-1">
    <v-toolbar-title-stub class="o-toolbar-title">
      <!----> <a class="white--text o-router-link"><img src="/assets/images/logo.svg" alt="Logo" width="40" height="40" class="hidden-sm-and-down o-toolbar-image"> <span>Oh My Vue</span></a></v-toolbar-title-stub>
    <!---->
    <v-spacer-stub></v-spacer-stub>
    <v-toolbar-items-stub>
      <v-btn-stub tag="button" activeclass="" to="/overview" text="true" type="button">Overview</v-btn-stub>
      <v-menu-stub opendelay="0" closedelay="0" contentclass="" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" offsety="true" origin="top left" transition="v-menu-transition">
        <v-list-stub tag="div" tile="true">
          <!---->
          <v-list-item-stub activeclass="" to="/hello-world" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>face</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Hello World</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/guide" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>loyalty</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Guide</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/crud-operations" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>business</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>CRUD Operations</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/controls" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>contacts</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Controls</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/data-table" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>fas fa-table</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Data Table</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/internationalization" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>language</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Internationalization</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/data-visualization" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>fas fa-chart-bar</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Data Visualization</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/authorization" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>verified_user</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Authorization</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/payment" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>far fa-credit-card</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Payment</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/messages" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>message</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Messages</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
          <v-list-item-stub activeclass="" to="/machine-learning" tag="div">
            <v-list-item-action-stub>
              <v-icon-stub>fas fa-cogs</v-icon-stub>
            </v-list-item-action-stub>
            <v-list-item-content-stub>
              <v-list-item-title-stub>Machine Learning</v-list-item-title-stub>
            </v-list-item-content-stub>
          </v-list-item-stub>
        </v-list-stub>
      </v-menu-stub>
    </v-toolbar-items-stub>
    <v-btn-stub tag="button" activeclass="" icon="true" type="button" aria-label="Theme">
      <v-icon-stub>M12,2A7,7 0 0,1 19,9C19,11.38 17.81,13.47 16,14.74V17A1,1 0 0,1 15,18H9A1,1 0 0,1 8,17V14.74C6.19,13.47 5,11.38 5,9A7,7 0 0,1 12,2M9,21V20H15V21A1,1 0 0,1 14,22H10A1,1 0 0,1 9,21M12,4A5,5 0 0,0 7,9C7,11.05 8.23,12.81 10,13.58V16H14V13.58C15.77,12.81 17,11.05 17,9A5,5 0 0,0 12,4Z</v-icon-stub>
    </v-btn-stub>
    <v-menu-stub opendelay="0" closedelay="0" contentclass="" bottom="true" left="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" openonclick="true" closeonclick="true" closeoncontentclick="true" maxheight="auto" origin="top left" transition="v-menu-transition">
      <v-list-stub tag="div" tile="true">
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-title-stub>English</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-title-stub>中文 (简体)</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-title-stub>中文 (正體)</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-title-stub>日本語</v-list-item-title-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-title-stub>한국어</v-list-item-title-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-menu-stub>
    <v-btn-stub tag="button" activeclass="" to="/authorization/login" icon="true" type="button">
      <v-icon-stub>M12,4A4,4 0 0,1 16,8A4,4 0 0,1 12,12A4,4 0 0,1 8,8A4,4 0 0,1 12,4M12,14C16.42,14 20,15.79 20,18V20H4V18C4,15.79 7.58,14 12,14Z</v-icon-stub>
    </v-btn-stub>
  </v-app-bar-stub>
  <v-content-stub tag="main">
    <div class="omv-content">
      <div class="omv-view">
        <v-fade-transition-stub mode="out-in" origin="top center 0">
          <router-view-stub name="default"></router-view-stub>
        </v-fade-transition-stub>
      </div>
      <v-footer-stub height="auto" tag="div" tile="true" class="pa-3 primary darken-1 white--text"><img src="/assets/images/logo.svg" alt="Logo" width="50" height="50">
        <v-layout-stub tag="div" fill-height="" justify-space-between="" align-center="" class="ml-3 mr-3">
          <div class="subheading"><a href="https://vuejs.org/" class="subheading pa-2 white--text o-router-link">Learn Vue</a></div>
          <div class="caption">
            Current Version: 2.6.11
          </div>
          <div class="caption text-xs-center">
            Powered by Oh My Vue ©2020. <br>
            Code licensed under an MIT-style License. <br>
            Documentation licensed under CC BY 4.0.
          </div>
        </v-layout-stub>
      </v-footer-stub>
    </div>
  </v-content-stub>
</v-app-stub>
`;
